src/Components/WebAssembly/WebAssembly/src/Hosting/WebAssemblyHostBuilder.cs

```diff
@@ -229,6 +237,8 @@ internal void InitializeDefaultServices()
             Services.AddSingleton<NavigationManager>(WebAssemblyNavigationManager.Instance);
             Services.AddSingleton<INavigationInterception>(WebAssemblyNavigationInterception.Instance);
             Services.AddSingleton(new LazyAssemblyLoader(DefaultWebAssemblyJSRuntime.Instance));
+            Services.AddSingleton<ComponentApplicationLifetime>();
+            Services.AddSingleton<ComponentApplicationState>(sp => sp.GetRequiredService<ComponentApplicationLifetime>().State);
```

Comment:
As per comment above, would be great to see this replace with a call like `host.Services.AddComponentStatePreservation()` in `Program.cs`.

Reply:
As per my comment [here](https://github.com/dotnet/aspnetcore/pull/29717#discussion_r579280920) are you ok if we leave it as-is for now and we revisit? The initialization logic is a bit "intricate" and don't want to add it to this PR